src/nvim/drawline.c

```diff
@@ -403,7 +403,7 @@ static int get_sign_attrs(buf_T *buf, linenr_T lnum, SignTextAttrs *sattrs, int
 ///
 /// @param[out] stcp  Status column attributes
 static void get_statuscol_str(win_T *wp, linenr_T lnum, int row, int startrow, int filler_lines,
-                              int cul_attr, int sign_num_attr, SignTextAttrs *sattrs,
```

Comment:
I think `sign_num_attr` is still needed. Consider the following test change:
```diff
diff --git a/test/functional/ui/sign_spec.lua b/test/functional/ui/sign_spec.lua
index a1683a32c..ae71707a8 100644
--- a/test/functional/ui/sign_spec.lua
+++ b/test/functional/ui/sign_spec.lua
@@ -156,6 +156,8 @@ describe('Signs', function()
         {0:~                                                    }|
                                                              |
       ]])
+      command('set statuscolumn=%s%l')
+      screen:expect_unchanged()
     end)
 
     it('highlights the cursorline sign with culhl', function()
```

Reply:
OK I'll check. Can you describe in words when `sign_num_attr` is supposed to be used lol? I thought it pertained to `set signcolumn=number`.

```c
  int sign_num_attr = 0;                // sign attribute for the number column
```

Reply:
But I see it is actually used in the `WL_NR` state which is why I guess I included it originally.

Reply:
It is used for `numhl` is signs.

Reply:
> It is used for `numhl` in signs.

Right, never realized while working on the code lol.